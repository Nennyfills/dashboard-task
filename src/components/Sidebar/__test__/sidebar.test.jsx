import { render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import SideBar from '..';

const history = createMemoryHistory();

let TagRender;
let Tag;

beforeEach(() => {
  // eslint-disable-next-line testing-library/no-render-in-setup
  TagRender = render(
    <Router location={history.location} navigator={history}>
      <SideBar />
    </Router>
  );
  Tag = (
    <Router location={history.location} navigator={history}>
      <SideBar />
    </Router>
  );
});

it('should take a snapshot', () => {
  const { asFragment } = TagRender;
  expect(asFragment(Tag)).toMatchSnapshot();
});

it('Should render an sidebar with class sidebar', () => {
  const tableClass = screen.getByTestId('sidebar');
  expect(tableClass).toHaveClass('sidebar');
});

it('Should render an sidebar with class sidebar--body', () => {
  const tableClass = screen.getByTestId('sidebar--body');
  expect(tableClass).toHaveClass('sidebar--body');
});

it('Should render the links in sidebar"', () => {
  const tableClass = screen.getAllByTestId('sidebar--link');
  expect(tableClass).toBeTruthy();
});
