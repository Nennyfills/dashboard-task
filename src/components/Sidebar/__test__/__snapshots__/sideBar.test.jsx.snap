// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <section
    class="sidebar"
    data-testid="sidebar"
  >
    <section
      class="sidebar--body"
      data-testid="sidebar--body"
    >
      <a
        class=""
        data-testid="sidebar--link"
        href="/"
      >
        <svg
          class="active "
        >
          chart.svg
        </svg>
      </a>
      <a
        class=""
        data-testid="sidebar--link"
        href="/block"
      >
        <svg
          class=""
        >
          block.svg
        </svg>
      </a>
      <a
        class=""
        data-testid="sidebar--link"
        href="/monitor"
      >
        <svg
          class=""
        >
          monitor.svg
        </svg>
      </a>
      <a
        class=""
        data-testid="sidebar--link"
        href="/reports"
      >
        <svg
          class=""
        >
          report.svg
        </svg>
      </a>
      <a
        class=""
        data-testid="sidebar--link"
        href="/switch"
      >
        <svg
          class=""
        >
          switch.svg
        </svg>
      </a>
    </section>
  </section>
</DocumentFragment>
`;
